tests/Rule/UseNonStaticAssertionsTest.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of DOCtor-RST.
+ *
+ * (c) Oskar Stark <oskarstark@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace App\Tests\Rule;
+
+use App\Rule\UseNonStaticAssertions;
+use App\Tests\RstSample;
+use App\Tests\UnitTestCase;
+use App\Value\NullViolation;
+use App\Value\Violation;
+use App\Value\ViolationInterface;
+
+final class UseNonStaticAssertionsTest extends UnitTestCase
+{
+    /**
+     * @test
+     *
+     * @dataProvider checkProvider
+     */
+    public function check(ViolationInterface $expected, RstSample $sample): void
+    {
+        self::assertEquals(
```

Comment:
```suggestion
        self::assertSame(
```
?

Reply:
TBH I copied from another test and `assertEquals` seems to be used almost everywhere 
BTW `assertSame` is breaking tests : 
`Failed asserting that two variables reference the same object.`
Using this seems a bit too restrictive

Reply:
Ok then